### PR TITLE
chore: upgrade and fix redux typings

### PR DESCRIPTION
**Pull request by tyleroooo**
useSelector -> useAppSelector
useDispatch -> useAppDispatch (this one technically is the same for now, but someday they'll type actions properly maybe)
createSelector -> createAppSelector

Also refactored how selectors that have parameters work: 
Any parameterized selector should be declared as a function which takes no arguments and returns a selector that takes arguments normally (a selector factory, if you will): 
```
export const getPositionDetails = () =>
  createAppSelector(
    [getExistingOpenPositions, getAssets, getPerpetualMarkets, (s, marketId: string) => marketId],
    (positions, assets, perpetualMarkets, marketId) => {
      const matchingPosition = positions?.find((position) => position.id === marketId);
      return matchingPosition
        ? getHydratedPositionData({ data: matchingPosition, assets, perpetualMarkets })
        : undefined;
    }
  );
```

Using them you must wrap the selector in a useMemo then call it normally: 
```
  const tradingRewardSelector = useMemo(getHistoricalTradingRewardsForCurrentWeek, []);
  const currentWeekTradingReward = useAppSelector(s => tradingRewardSelector(s, someArgumentHere));
```

But I made a helper hook for this:
```
const shouldRenderActions = useParameterizedSelector(
    getHistoricalTradingRewardsForCurrentWeek,
    someArgumentHere
  );
```

### CODE REVIEW

File: src/pages/token/rewards/TradingRewardsSummaryPanel.tsx [hunk 1]
```diff
@@ -18,17 +17,15 @@ import { Panel } from '@/components/Panel';
...
 
 import abacusStateManager from '@/lib/abacus';
 
 export const TradingRewardsSummaryPanel = () => {
   const stringGetter = useStringGetter();
   const { chainTokenLabel } = useTokenConfigs();
-  const canViewAccount = useSelector(calculateCanViewAccount);
-  const currentWeekTradingReward = useSelector(
-    getHistoricalTradingRewardsForCurrentWeek,
-    shallowEqual
```
[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/pages/token/rewards/TradingRewardsSummaryPanel.tsx:30]: Oops, this file I think was the one I made some kind of copy+paste mistake.

File: src/state/perpetualsSelectors.ts [hunk 2]
```diff
@@ -72,7 +72,7 @@ export const getLiveTrades = (state: RootState) => state.perpetuals.liveTrades;
 export const getCurrentMarketLiveTrades = (state: RootState) => {
   const liveTrades = getLiveTrades(state);
   const currentMarketId = getCurrentMarketId(state);
-  return currentMarketId ? liveTrades?.[currentMarketId] ?? [] : [];
+  return currentMarketId ? liveTrades?.[currentMarketId] ?? EMPTY_ARR : EMPTY_ARR;
```
[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/perpetualsSelectors.ts:75]: I think some could use it yeah, but didn't want this PR to get huge.

File: src/state/perpetualsSelectors.ts [hunk 3]
```diff
@@ -106,7 +106,7 @@ export const getHistoricalFundings = (state: RootState) => state.perpetuals.hist
 /**
  * @returns Historical funding data for the market the user is currently viewing
  */
-export const getCurrentMarketHistoricalFundings = createSelector(
+export const getCurrentMarketHistoricalFundings = createAppSelector(
   [getHistoricalFundings, getCurrentMarketId],
   (historicalFundings, currentMarketId) =>
     currentMarketId ? historicalFundings?.[currentMarketId] ?? [] : []
```
[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/perpetualsSelectors.ts:112]: Good call, I went ahead and fixed the rest in this file.

[the remark this answers is not in the document]
Comment (tyleroooo) [reply src/state/perpetualsSelectors.ts]: Actually there weren't many others of this pattern so I did it everywhere I could find.

**Review by aforaleka — APPROVED** [review]

**Pull request merged**